frame/executive/src/lib.rs

```diff
@@ -328,14 +335,15 @@ where
 
 			Self::initialize_block(block.header());
 
-			// any initial checks
-			Self::initial_checks(&block);
+			// any initial checks, and retrieve the index of the first signed extrinsic from the
+			// `extrinsics` vec if one exists.
+			let maybe_first_signed_index = Self::initial_checks(&block);
```

Comment:
comment and variable name are confusing, the first non-inherent can be an unsigned transaction.
```suggestion
			let maybe_first_non_inherent = Self::initial_checks(&block);
```